### PR TITLE
Feature/vue data row edit

### PR DESCRIPTION
**Pull request by ConnorStroomberg**
Optional way to edit table data using vue ui-form.

Off by default, can be turned on using dataexplorer settings

#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [x] User documentation updated
- [x] (If you have changed REST API interface) view-swagger.ftl updated
- [x] Test plan template updated
- [x] Clean commits
- [x] Added Feature/Fix to release notes
- [x] Integration tests run correctly

### CODE REVIEW

File: molgenis-data-row-edit/src/main/frontend/src/main.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+const {lng, fallbackLng, baseUrl, dataExplorerBaseUrl} = window.__INITIAL_STATE__
+
+const dataRowEditRouter = new Router({
+  mode: 'history',
+  base: baseUrl,
+  routes: [
+    {
+      path: '/:dataTableId/:rowId',
+      component: DataRowEdit
```
Comment (mark-de-haan) [inline molgenis-data-row-edit/src/main/frontend/src/main.js]: If you add props: true, you can set dataTableId and rowId as props inside your DataRowEdit component.
Saves you the hassle of doing this.$router.params... etc etc

File: molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/settings/DataExplorerSettings.java [hunk 2]
```diff
@@ -483,13 +480,26 @@ public void setEntityReports(String entityReportsStr)
...
+	public boolean isUseVueDataRowEdit()
+	{
+		return isTruefy(getBoolean(Meta.GENERAL_USE_VUE_DATA_ROW_EDIT));
+	}
+
+	/**
+	 * @param input nullable boolean
+	 * @return boolean indicating if input is true of false, is input is null false is returned
+	 */
+	private boolean isTruefy(@Nullable Boolean input)
```
Comment (mark-de-haan) [inline molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/settings/DataExplorerSettings.java]: I think you mean isTruthy 😋

File: molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/settings/DataExplorerSettings.java [hunk 3]
```diff
@@ -483,13 +480,26 @@ public void setEntityReports(String entityReportsStr)
...
 	}
+
+	public boolean isUseVueDataRowEdit()
+	{
+		return isTruefy(getBoolean(Meta.GENERAL_USE_VUE_DATA_ROW_EDIT));
+	}
+
+	/**
+	 * @param input nullable boolean
+	 * @return boolean indicating if input is true of false, is input is null false is returned
```
Comment (mark-de-haan) [inline molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/settings/DataExplorerSettings.java]: Eh I don't get this sentence

Suggestion:
Indicates if input is true **or** false
Returns false when input is null

**Comment by ConnorStroomberg** [comment]
Comment from demo: What if the user has plugin permissions on data-row-edit ? User should be warned ( or fallback to ‘DEFAULT’ mode)

**Comment by ConnorStroomberg** [comment]
move setting to data mapje in dataexplorer settings

**Comment by mark-de-haan** [comment]
Could you add the feature to the create row button as well?

**Comment by sidohaakma** [comment]
test this please

**Comment by sidohaakma** [comment]
test this please

**Pull request merged**